platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/common/DependencyCollectorFunctionExtractorAndRuntimeResolver.kt

```diff
@@ -162,11 +200,12 @@ class DependencyCollectorFunctionExtractorAndRuntimeResolver(
     override fun resolve(receiverClass: KClass<*>, schemaFunction: SchemaFunction, scopeClassLoader: ClassLoader): RuntimeFunctionResolver.Resolution {
         // We can't just use find receiverClass directly as a key because at runtime we get a decorated class with a different type
         // that extends the original class we extracted into the managedFunctions map, so we have to check the superClass
-        return typeHierarchyViaJavaReflection(receiverClass)
-            .firstNotNullOfOrNull(collectorDeclarationsByClass::get)
+        fun searchDeclarationsByClass(declarationsByClass: MutableMap<KClass<*>, Map<DataMemberFunction, DeclarativeRuntimeFunction>>): RuntimeFunctionResolver.Resolution.Resolved? = typeHierarchyViaJavaReflection(receiverClass)
+            .firstNotNullOfOrNull(declarationsByClass::get)
             ?.entries?.find { (fn, _) -> fn == schemaFunction }
             ?.value?.let(RuntimeFunctionResolver.Resolution::Resolved)
-            ?: RuntimeFunctionResolver.Resolution.Unresolved
+
+        return searchDeclarationsByClass(collectorDeclarationsByClass) ?: searchDeclarationsByClass(modifierDeclarationsByClass) ?: RuntimeFunctionResolver.Resolution.Unresolved
```

Comment:
💅 Given that `modifierDeclarationsByClass` and `collectorDeclarationsByClass` serve the same purpose and have the same shape, would it make sense to join them into one map?

Reply:
Also, there's likely more modifier declarations to be added soon (testFixtures).